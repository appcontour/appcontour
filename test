#!/bin/bash
#

l__ist=`ls tests/test.??`
if [ -n "$1" ]
then
  l__ist=tests/test.$1
fi

tmpf__ile=test$$.tmp
failed="0"

for f__ile in $l__ist
do
  i__nfo=$(grep "^#testname:" $f__ile | cut -f2- -d:)
  if [ -n "$i__nfo" ]; then i__nfo=" ($i__nfo)"; fi
  test=`echo $f__ile | cut -f2 -d\.`
  echo "Running test $test$i__nfo..."
  source $f__ile 2>/dev/null >$tmpf__ile
  exitcode=$?

  if [ -f ${f__ile}.out ]
  then
    if ! diff $tmpf__ile ${f__ile}.out
    then
      failed=$[ $failed + 1 ]
    fi
  else
    echo "...new test, copying out file"
    cp $tmpf__ile ${f__ile}.out
  fi
  rm $tmpf__ile
  echo "$exitcode" >$tmpf__ile
  if [ -f ${f__ile}.ec ]
  then
    if ! diff $tmpf__ile ${f__ile}.ec
    then
      failed=$[ $failed + 1 ]
    fi
  else
    echo "...new test, copying ec file"
    cp $tmpf__ile ${f__ile}.ec
  fi
  rm $tmpf__ile
done

if [ "$failed" -gt "0" ]
then
  echo $failed tests failed
  exit 1
fi

exit 0
