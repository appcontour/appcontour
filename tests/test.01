./contour applyrule n1 <examples/tre_sfere.morse
