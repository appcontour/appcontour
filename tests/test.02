#./contour applyrule n1 <examples/three_spheres.morse | ./contour applyrule n1 | ./contour applyrule n1 | ./contour applyrule n1
./contour applyrule k2 <examples/three_spheres.morse | ./contour applyrule k2 | ./contour applyrule k2 | ./contour applyrule k2
