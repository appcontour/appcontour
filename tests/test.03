./contour applyrule n1 <examples/tre_sfere.morse | ./contour applyrule n1 | ./contour applyrule n1
