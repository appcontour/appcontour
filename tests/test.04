./contour ishuffman -q examples/two_rings.morse
./contour iscontour -q examples/immersed/boysurface.morse
