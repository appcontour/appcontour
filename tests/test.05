./contour applyrule n2 <examples/sfera_dietro_toro.morse
./contour applyrule n3 <examples/sfera_davanti_toro.morse
./contour applyrule n4 <examples/due_anelli.morse | ./contour applyrule n2 | ./contour applyrule n3 | ./contour applyrule n1
./contour applyrule n4 <examples/toro2.morse
./contour applyrule N5 <examples/tre_sfere.morse | ./contour applyrule N5:6 | ./contour applyrule N1 | ./contour applyrule N1 | ./contour applyrule N1
