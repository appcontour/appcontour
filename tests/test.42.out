Canonifying ideal:
ideal(t) {
+4t-3;
+2t+3;
}
gives
Alexander polynomial (up to t -> 1/t):
+1;
==================
Canonifying ideal:
ideal(t) {
+4-7t+4t^2;
+4-6t+t^2+t^3;
+4-6t+t^2+t^3;
-4+5t+2t^2-2t^3;
}
gives
# *** Warning: result can be noncanonical ***
Alexander ideal generated by:
+5;
+1+2t+t^2;
==================
Canonifying ideal obtained from kanenoby 10 -2
gives
==================
Canonify ideal:
ideal(t) {
+332177328;
+12+68240532t;
+1353876-1607472t;
+1+16057617t+55111t^2;
+69022+140468496t+544144t^2;
-477317+19746927t-12953t^2;
+5177-82979183t+709211t^2+616t^3;
-4513+652924t+26830052t^2+t^3;
+2589-58686413t+787863t^2+17104t^3;
}
gives
# *** Warning: result can be noncanonical ***
Alexander ideal generated by:
+6;
+1+3t+t^2;
==================
The next two should coincide
# *** Warning: result can be noncanonical ***
Alexander ideal generated by:
+4;
+1-t;
==================
# *** Warning: result can be noncanonical ***
Alexander ideal generated by:
+4;
+1-t;
==================
Result of some ideals, see source for details:
# *** Warning: result can be noncanonical ***
Alexander ideal generated by:
+9;
+3-3t;
+1+t+t^2;
==================
# *** Warning: result can be noncanonical ***
Alexander ideal generated by:
+6;
+1+t;
==================
# *** Warning: result can be noncanonical ***
Alexander ideal generated by:
+9;
+3+3t+3t^2;
==================
# *** Warning: result can be noncanonical ***
Alexander ideal generated by:
+9-9t^3;
+3-3t-3t^3+3t^4;
+1+t+t^2-t^3-t^4-t^5;
==================
# *** Warning: result can be noncanonical ***
Alexander ideal generated by:
+9-9t^3;
+3-3t-3t^3+3t^4;
+1+t+t^2-t^3-t^4-t^5;
==================
knots 12a_0396 and 12a_0435 have unusual second elementary ideal
Computing second Alexander ideal for knot with dtcode: [4 10 16 20 2 22 24 6 14 8 18 12]
# *** Warning: result can be noncanonical ***
Alexander ideal generated by:
+9;
+3+3t;
-1+4t-t^2;
Computing second Alexander ideal for knot with dtcode: [4 10 18 14 2 22 16 8 20 6 24 12]
# *** Warning: result can be noncanonical ***
Alexander ideal generated by:
+2-2t+2t^2;
+1+t^2+t^4;
==================
